### ISSUE TITLE
Inquiry about the files on huggingface

### ISSUE THREAD

**Issue by beidiz** [aimagelab/ReT-2#2]
Hi, I noticed there are some files at "jsonl/m2kr/test/xxx_test.json" already contain related passages in "passage_text". Does that mean the most related passages? Can I directly use them to test the RAG generation performance with ReT-2's output?

**Comment by dcaffo98** [comment]
Hi @beidiz!
Yes, `passage_text` is (or are) the most related passage(s) (_i.e._ the ground-truth) that ideally should be retrieved from the associated query, according to the original M2KR dataset. 
Concerning RAG, you should not use them, and instead you should use ReT, or your preferred retriever, to retrieve a relevant passage. Then, you may use `passage_text` to compute the `Pseudo-Recall@k`[1] metric to assess the quality of your retriever.
At most, you can feed your answer generator with data from `passage_text` to evaluate its VQA performance in an "oracle" setting, where you are always given the passage containing the answer.

[1] Lin, Weizhe, et al. "PreFLMR: Scaling Up Fine-Grained Late-Interaction Multi-modal Retrievers." In ACL 2024

**Comment by beidiz** [comment]
Thanks @dcaffo98 for your prompting reply! May I ask could you please upload this kind of retrieved text using ReT or ReT-2 for testing the generation accuracy? 
Another question is that I tried to use zero-shot and RAG with Qwen2.5-VL-7B to get the output on OKVQA. For zero-shot, it shows around 61% acc, while for RAG, it only shows around 48% acc. I used both "passage_text" and the text retrieved with ReT-2, and the performance are both very low. I am quite confused. For the setting, I used top-3 texts, and the prompts are the same as the ones in src/rag/prompts.py. I also tried with different temperatures and beams and the trends are the same. 
Table VII shows text retrieved from ReT-2 can help improve the VQA accuracy score on E-VQA and InfoSeek. I was wondering whether it works for OKVQA?

**Comment by dcaffo98** [comment]
> May I ask could you please upload this kind of retrieved text using ReT or ReT-2 for testing the generation accuracy?

We'll upload them as soon as we have some extra time. However, you should be able to recreate the results with the released ReT-2 checkpoints and by following [these steps](https://github.com/aimagelab/ReT-2?tab=readme-ov-file#rag-vqa).

> For zero-shot, it shows around 61% acc, while for RAG, it only shows around 48% acc. I used both "passage_text" and the text retrieved with ReT-2, and the performance are both very low.

The fact that it does not improve even with the ground-truth passage looks strange, even though I imagine that most questions in OK-VQA can be answered by Qwen2.5-VL-7B without the need for external retrieval [1]. That's why researchers created datasets such as InfoSeek and E-VQA.

[1] Chen, Yang, et al. "Can Pre-trained Vision and Language Models Answer Visual Information-Seeking Questions?." In EMNLP 2023

**Comment by beidiz** [comment]
Thanks, I will try with InfoSeek and E-VQA.

**Comment by beidiz** [comment]
Hi, may I ask a follow-up question? I noticed in this page https://huggingface.co/datasets/aimagelab/ReT-M2KR/tree/main/jsonl/m2kr/test, there are many processed datasets. I was wondering whether the key "passages" is the truth to be compared with when calculating R@5 in the paper? In other words, for one sample, if the top-5 chunks we retrieve contain one chunk in the "passages", we regard R@5 as 1 for that data sample.

**Comment by dcaffo98** [comment]
Hi @beidiz 
To compute `R@k`, you have to rely on the field [`passage_id`](https://github.com/aimagelab/ReT-2/blob/5118e93cd75c8e4be6d669ae5575a971f88a5f6b/src/evaluate.py#L321). That is a list of unique passage identifiers ("pids"), and if the pid of at least one of the top-k retrieved passages falls into that list, then that query counts as one "hit" for the purpose of`R@k`.
Note that, according to recent works, we use the term "recall at k", i.e., `R@k`, whereas that is not a true recall expressed as the ratio between the true positives and the (true positives + false negatives). It measures the share of queries for which a relevant item is found within the top-k retrieved items, also known as "hit rate at k".

**Comment by beidiz** [comment]
Hi @dcaffo98, thanks a lot for the precise explanation! It perfectly clarifies the definition of R@k and the role of passage_id. Then, can I assume that the `passage_text` corresponding to the `passage_id` contains the related contents to the question?

**Comment by dcaffo98** [comment]
Yes @beidiz , you can.

**Comment by beidiz** [comment]
@dcaffo98 Thanks for all your patience to answer my questions!
